src/TodoContext.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useMemo, useState } from 'react';
+import * as todoService from './api/todos';
+import { Todo } from './types';
+import { DELETE_ERROR, UPDATE_ERROR, USER_ID } from './utils/constants';
+import { pickCompletedTodos } from './utils/pickUncompletedTodos';
+
+interface TodoContextType {
+  todoItems: Todo[];
+  setTodoItems: React.Dispatch<React.SetStateAction<Todo[]>>;
+  tempTodo: Todo | null;
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  processingTodoIds: number[],
+  setProcessingTodoIds: (loadingTodoIds: number[]) => void,
+  errorMessage: string;
+  setErrorMessage: (errorMessage: string) => void;
+  deleteTodo: (id: number) => void;
+  completedTodos: Todo[];
+  clearAllCompleted: () => void;
+  uncompletedTodosLength: number;
+  handleTodoUpdate: (todo: Todo, todoTitle: string) => void;
+  handleStatusChange: (todo: Todo, status: boolean) => void;
+  setStatusForAll: () => void;
+}
+
+const todoContext: TodoContextType = {
+  todoItems: [],
+  setTodoItems: () => {},
+  tempTodo: null,
+  setTempTodo: () => {},
+  processingTodoIds: [],
+  setProcessingTodoIds: () => {},
+  errorMessage: '',
+  setErrorMessage: () => {},
+  deleteTodo: () => {},
+  completedTodos: [],
+  clearAllCompleted: () => {},
+  uncompletedTodosLength: 0,
+  handleTodoUpdate: () => {},
+  handleStatusChange: () => {},
+  setStatusForAll: () => {},
+};
+
+export const TodoContext = React.createContext(todoContext);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todoItems, setTodoItems] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+
+  const deleteTodo = (todoId: number) => {
+    setProcessingTodoIds(prevTodoIds => [...prevTodoIds, todoId]);
+    todoService.deleteTodo(todoId)
+      .then(() => setTodoItems(currentTodos => currentTodos
+        .filter(todo => todo.id !== todoId)))
+      .catch((error) => {
+        setErrorMessage(DELETE_ERROR);
+        throw error;
```

Comment:
Remember if you throw an error you have to catch it on the next step, in another case you don't need it.

In your case I think you don't need to throw it as you set error message and notify the user